### DISCUSSION TITLE
Use 'Artist Sort Order' from file metadata when sorting by Artist

### DISCUSSION THREAD

**Discussion by codeblue11** [music-assistant/support#1570] [Feature requests and ideas]
Currently when you sort by artist it looks like Music Assistant sorts alphabetically by the first letter of the artist name. Often I would prefer to sort artists by first letter in the last name (as I think is common in the US).

If there is 'Artist Sort Order' in the metadata tag please use that for sorting by artist. Then the user can tag the files appropriately so it's sorted how they would prefer.

**Comment by OzGav** [comment]
Noted. If we revisit this part of the code we will look into this for you.

**Reply by marcelveldt** [reply]
We already use the sort name for sort actions, for instance the sort name of "the Beatles" is "Beatles, the"
Sorting by first letter of the last name is not common, that would turn "Ed Sheeran" into "Sheeran, Ed" ?

Under the hood we already use the sort name so some day we could expose this info and let the user edit it.

**Comment by codeblue11** [comment]
> We already use the sort name for sort actions, for instance the sort name of "the Beatles" is "Beatles, the" Sorting by first letter of the last name is not common, that would turn "Ed Sheeran" into "Sheeran, Ed" ?


Correct, I would prefer to sort by Sheeran, Ed. Another good example is *NSYNC is the Artist, but NSYNC is the Artist Sort Order when tagged using MusicBrainz Picard. I would prefer to use Artist Sort Order here because currently in Music Assistant *NSYNC shows up as the first artist when I would prefer to have it sorted with the N's.

**Reply by OzGav** [reply]
We will probably monitor this discussion and see how many upvotes its gets to steer the final decisions with regards to this request. Thanks for raising it.

**Reply by marcelveldt** [reply]
We should already use the sort order from musicbrainz picard - we actually read that sort name tag. If that i snot working correctly, we have a bug

**Reply by OzGav** [reply]
It would completely change the way artists are sorted...FYI... Currently "The" is dropped but Artists that are not groups are sorted by first name.

Edit: Not that it is the reference that should necessarily be used but searching on Spotify suggests they are sorting by first name.

Edit 2: I'm not arguing either way just providing some background info

**Reply by codeblue11** [reply]
> It would completely change the way artists are sorted...FYI... Currently "The" is dropped but Artists that are not groups are sorted by first name.


Yes, this seems to be how it works, not by Sort Order. It just drops "The" then sorts by the first character.

Maybe sorting by last name is an American thing? I know it's very common here, record stores almost always sort by Last Name, First Name for solo artists.

The good thing is that if the individual user doesn't like the way it's sorted, they would be able to change it however they like by changing the Artist Sort Order tag. Currently there is no option to change it.

**Reply by codeblue11** [reply]
[FYI, preference seems to be pretty mixed according to this Reddit poll.](https://www.reddit.com/r/Cd_collectors/comments/1647poe/do_you_sort_artist_by_first_or_last_names/)

**Reply by OzGav** [reply]
Yeah hard to keep everyone happy 😁

**Comment by OzGav** [comment]
The feature request has been implemented successfully. As such this feature request will be closed and locked. If there are enhancements desired then open a new [feature request](https://github.com/orgs/music-assistant/discussions/categories/feature-requests-and-ideas). If there is a question about how this works then check the [documentation](https://beta.music-assistant.io/), or ask a question in the [Q&A](https://github.com/orgs/music-assistant/discussions/categories/q-a) area or on Discord. If there are problems then ask a question in the [Q&A](https://github.com/orgs/music-assistant/discussions/categories/q-a) area or on Discord or open an [issue](https://github.com/music-assistant/support/issues). Thanks for your interest in seeing this idea being brought to life.
